src/rtc/geogrid.py

```diff
@@ -268,13 +270,60 @@ def get_point_epsg(lat, lon):
         raise ValueError(err_str)
 
 
+def _check_pixel_spacing(y_spacing_positive, x_spacing, epsg, product_str):
+    '''
+    Verify pixel spacing
 
+    Parameters
+    ----------
+    y_spacing_positive: float
+        Y/latitude pixel spacing (absolute value)
+    x_spacing: float
+        X/longitude pixel spacing
+    epsg: int
+        EPSG code
+    product_str: string
+        Product type string (e.g., "Mosaic" or "Bursts")
 
+    Returns
+    -------
+    y_spacing: float
+        Y/latitude pixel spacing
+    x_spacing: float
+        X/longitude pixel spacing
+    '''
+    # Check spacing in X direction
+    if x_spacing is not None and x_spacing <= 0:
+        err_str = f'{product_str} pixel spacing in X/longitude direction must'
+        err_str += f' be positive (x_spacing: {x_spacing})'
+        logger.error(err_str)
+        raise ValueError(err_str)
+    elif x_spacing is None and not epsg:
+        x_spacing = None
+    elif x_spacing is None and epsg == 4326:
+        x_spacing = 0.00027
+    elif x_spacing is None:
+        x_spacing = 30
```

Comment:
```suggestion

    if x_spacing is None and epsg:
        x_spacing = 30
    elif x_spacing is None and epsg == 4326:
        x_spacing = 0.00027
```
Early return for the previous if statement in line 296 . Also no need to assign `None` when `x_spacing` is already `None`.

Reply:
Thank you, @seongsujeong . I think the logic above will fail if `epsg` is `4326` (we'd be assigning `30` to `x_spacing`), but I got the idea. Thank you!